include/tvm/runtime/packed_func.h

```diff
@@ -336,8 +337,11 @@ class TVMArgs {
 inline const char* ArgTypeCode2Str(int type_code);
 
 // macro to check type code.
+#define TVM_ICHECK_TYPE_CODE(CODE, T) \
+  ICHECK_EQ(CODE, T) << " expected " << ArgTypeCode2Str(T) << " but got " << ArgTypeCode2Str(CODE)
```

Comment:
Given the macro is only used inside the packed_func, let us only keep one version. TVM_CHECK_TYPE_CODE should be fine for this case. (feel free to use ICHECK internally)